### PR TITLE
DSD 1829: Adding responsive tooltip example to Card docs

### PR DESCRIPTION
**Pull request by 7emansell**
Fixes JIRA ticket [DSD-1829](https://newyorkpubliclibrary.atlassian.net/browse/DSD-1829).

## This PR does the following:

- Adds code example of `Tooltip` workaround on a fully clickable `Card` with responsive offset values 
  - Don't think this requires a changelog update as it's really part of the work from #1652

## How has this been tested?

Locally

### Accessibility Checklist

- [x] Checked Storybook's "Accessibility" tab for color contrast and other issues.
- [x] The feature works with keyboard inputs including tabbing back and forward and pressing space, enter, arrow, and esc keys.
- [x] For hidden text or when `aria-live` is used, a screenreader was used to verify the text is read.
- [x] For features that involve UI updates and focusing on DOM `ref`s, focus management was reviewed.
- [x] The feature works when the page is zoomed in to 200% and 400%.

### Checklist:

- [x] I have updated the Storybook documentation and changelog accordingly.
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [x] Review the Vercel preview deployment once it is ready.

[DSD-1829]: https://newyorkpubliclibrary.atlassian.net/browse/DSD-1829?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Card/Card.stories.tsx [hunk 1]
```diff
@@ -767,37 +767,29 @@ export const CardFullClickTurbineExample: Story = {
...
-    -350, -450,
-  ]);
-  const [tooltipText, setTooltipText] = useState("Default offset");
-
+  const [offset, setOffset] = useState<[number, number]>([0, 0]);
+  const cardRef = useRef<HTMLDivElement>(null);
+  const getOffset = () => {
+    if (cardRef.current) {
+      const image = cardRef.current.children[0] as HTMLElement;
+      const percentageHeightOffset = image.offsetHeight * 1.01;
```
Comment (jackiequach) [inline src/components/Card/Card.stories.tsx:775]: How are these values calculated and should a small code snippet be added to the Storybook docs?

Comment (bigfishdesign13) [reply src/components/Card/Card.stories.tsx:775]: Yes, let's include a code example with this functional example.  It works great.  Now let's show how it was accomplished.

**Comment by 7emansell** [comment]
I really cannot get the formatting in the code block to work does anyone have tips?

File: src/components/Card/Card.mdx [hunk 2]
```diff
@@ -246,8 +246,41 @@ This example can be found in the [Turbine homepage](https://nypl-ds-test-app.ver
...
-If you need a tooltip within a clickable `Card`, wrap the entire `Card` component inside the `Tooltip` and use the `placement` and `offset` props to move the `Tooltip` to display next to the intended reference.
-Keep in mind that `offset` is static pixel values, so check where the `Tooltip` displays on all breakpoints. You may need to keep track of the offset values through state, and update them conditionally with the `useNYPLBreakpoints` hook.
+If you need a tooltip within a clickable `Card`, wrap the entire `Card` component inside the `Tooltip` and use the `Tooltip`'s `placement` and `offset` props to move the `Tooltip` to display next to the intended reference.
+Keep in mind that `offset` is static pixel values, so check where the `Tooltip` displays on all breakpoints.
+
+You may need to keep track of the offset values through state, and update accordingly– see below example.
+
+<Source
+  code={`
+  const [offset, setOffset] = useState<[number, number]>([0, 0]);
```
Comment (EdwinGuzman) [inline src/components/Card/Card.mdx:256]: Re: formatting issues
- dedent the code by one tab
- update the language to "tsx"

File: src/components/Card/Card.mdx [hunk 3]
```diff
@@ -246,8 +246,41 @@ This example can be found in the [Turbine homepage](https://nypl-ds-test-app.ver
...
+  };
+  useEffect(() => {
+    setTimeout(getOffset, 0);
+    window.addEventListener("resize", getOffset);
+    return () => {
+        window.removeEventListener("resize", getOffset);
+    };}, 
+  []);
+
+<Tooltip offset={offset} content={"Tooltip text"}>
```
Comment (EdwinGuzman) [inline src/components/Card/Card.mdx:276]: Either above this line add a comment as `// ... other code` or include a `return` statement here so it appears like valid code. The intention is for a code snippet so I tend to do the former.

**Review by EdwinGuzman — CHANGES_REQUESTED** [review]
This works well but I noticed the image updates as I change the width of the window. This seems to be an existing issue and most likely rerendering because of the state and update functions being called. Add the card's image value `getPlaceholderImage("smaller")` (line 796) in a ref so only one image href is set for this example.

File: src/components/Card/Card.stories.tsx [hunk 4]
```diff
@@ -793,7 +793,7 @@ function FullClickWithTooltipExample() {
         imageProps={{
           alt: "Alt text",
           aspectRatio: "twoByOne",
-          src: getPlaceholderImage("smaller"),
+          src: "https://images.nypl.org/index.php?id=swope_243025&t=r",
```
Comment (EdwinGuzman) [inline src/components/Card/Card.stories.tsx:796]: Yes, this also works.

File: src/components/Card/Card.mdx [hunk 5]
```diff
@@ -246,8 +246,44 @@ This example can be found in the [Turbine homepage](https://nypl-ds-test-app.ver
...
+      setOffset([-percentageWidthOffset, -percentageHeightOffset]);
+    }
+  };
+useEffect(() => {
+  setTimeout(getOffset, 0);
+  window.addEventListener("resize", getOffset);
+  return () => {
+      window.removeEventListener("resize", getOffset);
+  };}, 
+[]);
```
Comment (EdwinGuzman) [inline src/components/Card/Card.mdx]: Oh, I just noticed this. Is this another formatting issue you flagged before?

**Review by EdwinGuzman — APPROVED** [review]
The formatting isn't prefect but I don't think it's a big deal. It's still valid code and we've had issues with MDX in these code snippets before -- it's a pain.

File: src/components/Card/Card.mdx (hunk 5 above)
Comment (7emansell) [reply src/components/Card/Card.mdx]: Just in general the indenting was wrong

**Review by bigfishdesign13 — APPROVED** [review]
Very nice.

**Pull request merged**